### PR TITLE
qt/kivy: receive tab: add wallet.ReceiveRequestHelp and refactor

### PR DESCRIPTION
**Pull request by SomberNight**
This PR refactors the Qt and kivy receive request displays in terms of error handling, and moves the common code into `wallet.py`.

"Error-like" help texts are split into two categories:
- errors, such as:
    - `"You must be online to receive Lightning payments."`
    - `"This request cannot be paid on-chain"`
- warnings, such as:
    - `"This address has already been used. For better privacy, do not reuse it for new payments."`

Both the kivy and the Qt GUIs handle these "error-like" scenarios now. If there is an error, the QR code (or corresponding text) is not shown, only the error. If there is a warning, the QR code is still shown, and the warning is also displayed somehow (up to GUI).

Previously the Qt GUI had complex GUI-specific logic for this, and kivy also had some primitive handling, with some duplication between them. Now the duplication is minimised and non-GUI-specific logic is moved to `wallet.py`.

-----

Examples:

<details>
 <summary>kivy warning</summary>

Capture1
</details>

<details>
 <summary>kivy error</summary>

Capture2
</details>

<details>
 <summary>qt warning</summary>

Capture3
</details>

<details>
 <summary>qt error</summary>

Capture4
Capture5
</details>

### CODE REVIEW

**Comment by ecdsa** [comment]
it looks good.
perhaps in Kivy we could toggle between qr code and text display, like on desktop, when the text is tapped.
(we currently dim the contrast when it is tapped, not sure if that is very useful)

**Comment by ecdsa** [comment]
One minor issue is that the help text may contain "you may have that capacity if you rebalance your channels", even though the kivy interface does not offer that option.

File: electrum/gui/kivy/uix/dialogs/request_dialog.py [hunk 1]
```diff
@@ -159,25 +179,47 @@ def on_mode(self, instance, x):
...
+        elif self.mode == self.MODE_LIGHTNING:
+            if help_texts.ln_is_error:
+                self.error_text = help_texts.ln_help
+            else:
+                self.data = lnaddr
+                self.warning = help_texts.ln_help
+        else:
+            raise Exception(f"unexpected {self.mode=!r}")
+        if self.warning:
+            self.warning = _('Warning') + ': ' + self.warning
```
Comment (ecdsa) [inline electrum/gui/kivy/uix/dialogs/request_dialog.py:222]: Note that self.warning is a kivy property. 
The GUI will be updated everytime it is changed.
This is why the original code was setting a local variable first.

Comment (ecdsa) [reply electrum/gui/kivy/uix/dialogs/request_dialog.py:222]: The same applies to `self.error_text`, btw

File: electrum/gui/kivy/uix/dialogs/request_dialog.py [hunk 2]
```diff
@@ -159,25 +179,47 @@ def on_mode(self, instance, x):
             self.ids.qr.opacity = 1
         else:
             self.ids.qr.opacity = 0
```
Comment (ecdsa) [inline electrum/gui/kivy/uix/dialogs/request_dialog.py:181]: changing the qr opacity is not needed anymore, I guess.

**Comment by ecdsa** [comment]
merged manually after rebase

File: electrum/gui/kivy/uix/dialogs/request_dialog.py (hunk 1 above)
Comment (SomberNight) [reply electrum/gui/kivy/uix/dialogs/request_dialog.py:222]: oh right, indeed

File: electrum/gui/kivy/uix/dialogs/request_dialog.py (hunk 2 above)
Comment (SomberNight) [reply electrum/gui/kivy/uix/dialogs/request_dialog.py:181]: right. removed in follow-up https://github.com/spesmilo/electrum/commit/13570a465c0c017803f45cb4ad96f2b349ffc2bb

**Pull request closed without merging**